core/trino-parser/src/test/java/io/trino/sql/parser/TestSqlParser.java

```diff
@@ -4350,6 +4350,60 @@ public void testTableFunctionTableArgumentAliasing()
                         ImmutableList.of())));
     }
 
+    @Test
+    public void testCopartitionInTableArgumentAlias()
+    {
+        // table argument 'input' is aliased. The alias "copartition" is illegal in this context.
+        assertThatThrownBy(() -> SQL_PARSER.createStatement(
+                "SELECT * " +
+                        "FROM TABLE(some_ptf( " +
+                        "input => TABLE(orders) AS copartition(a, b, c)))",
```

Comment:
It should be allowed in this case since it's not ambiguous. We should only declare ambiguity when the `AS` keyword is missing. The error should message indicate that the ambiguity can be resolved by adding such keyword.

Reply:
@martint done. Please take a look.